File: mail/common/src/main/java/com/fsck/k9/mail/MessagingException.java

package com.fsck.k9.mail;

public class MessagingException extends Exception {
    public static final long serialVersionUID = -1;

    private boolean permanentFailure = false;

    public MessagingException(Throwable cause) {
        super(cause);
    }

    public MessagingException(String message) {
        super(message);
    }

    public MessagingException(String message, boolean perm) {
        super(message);
        permanentFailure = perm;
    }

    public MessagingException(String message, Throwable throwable) {
        super(message, throwable);
    }

    public MessagingException(String message, boolean perm, Throwable throwable) {
        super(message, throwable);
        permanentFailure = perm;
    }

    public boolean isPermanentFailure() {
        return permanentFailure;
    }

}

File: mail/common/src/main/java/com/fsck/k9/mail/MessagingException.kt
package com.fsck.k9.mail

open class MessagingException : Exception {
    var isPermanentFailure: Boolean = false
        private set

    constructor(cause: Throwable?) : super(cause)

    constructor(message: String?) : super(message)

    constructor(message: String?, perm: Boolean) : super(message) {
        this.isPermanentFailure = perm
    }

    constructor(message: String?, throwable: Throwable?) : super(message, throwable)

    constructor(message: String?, perm: Boolean, throwable: Throwable?) : super(message, throwable) {
        this.isPermanentFailure = perm
    }

    companion object {
        val serialVersionUID: Long = -1
    }
}
